src/Parser.php

```diff
@@ -40,14 +40,14 @@ public function hasValidSyntax(string $pattern) : bool
 
     public function parse(string $pattern) : Pattern\TypePattern
     {
-        $AST = $this->getAST($pattern);
-        $pattern = new Pattern\TypePattern((string) $AST->getType());
```

Comment:
`$pattern` is already used as a string argument and reusing variable name with a different type is not recommended.